File: jdi tests/jdi-uitest-webtests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/pages/DatesPage.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.pages;

import com.epam.jdi.uitests.core.interfaces.common.IFileInput;
import com.epam.jdi.uitests.core.interfaces.common.ILabel;
import com.epam.jdi.uitests.web.robot.RFileInput;
import com.epam.jdi.uitests.web.selenium.elements.common.DatePicker;
import com.epam.jdi.uitests.web.selenium.elements.composite.WebPage;
import org.openqa.selenium.support.FindBy;

/**
 * Created by Maksim_Palchevskii on 8/17/2015.
 */
public class DatesPage extends WebPage {
    @FindBy(css = "#datepicker input")
    public DatePicker datepicker;
    @FindBy(css = "[data-provides=fileinput]")
    public IFileInput imageInput;
    @FindBy(css = "[data-provides=fileinput]")
    public RFileInput rImageInput;
    @FindBy(css = "[class=filename] span")
    public ILabel uploadedFileName;


}

File: Python/Test/jdi_uitests_webtests/main/page_objects/pages/dates_page.py
from JDI.web.os_action.r_file_input import RFileInput
from JDI.web.selenium.elements.api_interact.find_element_by import By
from JDI.web.selenium.elements.common.link import Link
from JDI.web.selenium.elements.composite.web_page import WebPage
from Test.jdi_uitests_webtests.main.page_objects.sections.contact_form import ContactForm


class DatesPage(WebPage):

    def __init__(self, url, title):
        super(DatesPage, self).__init__(url=url, title=title)

    r_image_input = RFileInput(By.css("[data-provides=fileinput]"))

    uploaded_file_name = Link(By.css("[class=filename] span"))

    contact_form = ContactForm(By.css("main form"))